tools/topology2/buffer.conf

```diff
@@ -0,0 +1,144 @@
+
+#
+# Common pipeline buffer
+#
+# A generic buffer component. All attributes defined herein are namespaced
+# by alsatplg to "buffer.attribute_name"
+#
+# Usage: this component can be used by declaring in the "widgets" field of
+# a parent object. i.e.
+#
+# widgets [
+# 	"buffer.M.N" {
+#		size	"384"
+#		caps	"101"
+#		underrun_permitted	"true"
+#	}
+# ]
+#
+# Where M is pipeline ID and N is a unique integer in the parent object.
+
+DefineSectionWidget."buffer" {
+
+	############### Boiler Plate Start ####################
+
+	#
+	# Argument used to construct pipeline ID (widget index)
+	#
+
+	@args [ INDEX ]
+
+	@args.INDEX {
+		type integer
+		default {
+			name buffer.index
```

Comment:
copy and paste from dmix code. I'm not keen on defining pipeline/component naming/index rules in the conf file so this can be removed.

Reply:
copied from dmix example - cant be dropped. I don't like setting the naming rules here where they can be easily broken. Lets do this in the compiler.

Reply:
oh... This is a default value for the buffer's `index` attribute, not related to the INDEX argument. It seems to be in the wrong scope then. And it seems to me, that you're using this `default` statement to define the buffer name, which isn't defined in the topology file, i.e. it would rely on some hard-coded compiler behaviour.